src/care/care_inst.cpp

```diff
@@ -10,8 +10,8 @@
 
 #define GPU_ACTIVE
 
-#include "care/algorithm.h"
-#include "care/KeyValueSorter.h"
+#include "care/algorithm_impl.h"
```

Comment:
I added the decl header as part of the impl headers for convenience.  This means that the instance body file does not actually need to include the regular header.  I'm not sure if that is a good or bad thing.

If you think it is better, I can remove the include in the impl files and require it to be included explicitly (though it might not show up as an error if inlining is enabled).

Reply:
I guess I'm fine with it either way now that I understand a bit better. Feel free to leave as is.